To: vim_dev@googlegroups.com
Subject: Patch 7.3.360
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.360
Problem:    Interrupting the load of an autoload function may cause a crash.
Solution:   Do not use the hashitem when not valid. (Yukihiro Nakadaira)
Files:	    src/eval.c


*** ../vim-7.3.359/src/eval.c	2011-11-30 14:57:26.000000000 +0100
--- src/eval.c	2011-11-30 15:06:53.000000000 +0100
***************
*** 19589,19597 ****
  	 * worked find the variable again.  Don't auto-load a script if it was
  	 * loaded already, otherwise it would be loaded every time when
  	 * checking if a function name is a Funcref variable. */
! 	if (ht == &globvarht && !writing
! 			    && script_autoload(varname, FALSE) && !aborting())
  	    hi = hash_find(ht, varname);
  	if (HASHITEM_EMPTY(hi))
  	    return NULL;
      }
--- 19589,19602 ----
  	 * worked find the variable again.  Don't auto-load a script if it was
  	 * loaded already, otherwise it would be loaded every time when
  	 * checking if a function name is a Funcref variable. */
! 	if (ht == &globvarht && !writing)
! 	{
! 	    /* Note: script_autoload() may make "hi" invalid. It must either
! 	     * be obtained again or not used. */
! 	    if (!script_autoload(varname, FALSE) || aborting())
! 		return NULL;
  	    hi = hash_find(ht, varname);
+ 	}
  	if (HASHITEM_EMPTY(hi))
  	    return NULL;
      }
*** ../vim-7.3.359/src/version.c	2011-11-30 14:57:26.000000000 +0100
--- src/version.c	2011-11-30 15:17:28.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     360,
  /**/

-- 
hundred-and-one symptoms of being an internet addict:
217. Your sex life has drastically improved...so what if it's only cyber-sex!

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
